### PR TITLE
Update Alertmanager Configmap when odh-parameters secret is updated

### PR DESCRIPTION
**Pull request by anishasthana**
Signed-off-by: Anish Asthana <anishasthana1@gmail.com>

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message
- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-1974
- [x] The Jira story is acked
- [x] An entry has been added to the latest build document in [Build Announcements Folder](https://drive.google.com/drive/folders/1sgkK1WZgGo9CXsLizNe0GbAzVKuSKrZL).
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious)

Test Instructions:
1) Deploy RHODS oeprator normally via the live build image.
2) Once all pods are up and running, look at the `prometheus` pod in `redhat-ods-monitoring`
3) Note the value of the `alertmanager` annotation
4) In the same namespace, look at the `to` emails next to `user-notifications` in the `alertmanager` configmap.
5) Now, update the `notification-email` key-value pair withing `addon-managed-odh-parameters` secret in the `redhat-ods-operator` namespace to have a new email.
6) After a few seconds ( you can look at pod logs to see when it happen), you will notice the prometheus pod restart
7) Look at the same annotation as before, it should have a different value now (you don't need to decrypt the base64 stuff)
8) The alertmanager configmap should have been updated too to be pointed at the new user-notification email

Related PR: https://github.com/red-hat-data-services/odh-deployer/pull/226

### CODE REVIEW

File: pkg/controller/kfdef/kfdef_controller.go [hunk 1]
```diff
@@ -1,44 +1,48 @@
...
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/kubernetes/pkg/apis/apps"
```
Comment (VaishnaviHire) [inline pkg/controller/kfdef/kfdef_controller.go:37]: This should be updated to `appsv1 "k8s.io/api/apps/v1"`

**Comment by anishasthana** [comment]
Live build is available at quay.io/anishasthana/rhods-operator-live-catalog:1.100.0-1974.

File: pkg/controller/kfdef/kfdef_controller.go [hunk 2]
```diff
@@ -695,3 +743,54 @@ func removeCsv(	c client.Client, r *rest.Config) error{
...
+
+	return fmt.Sprintf("'%s'", string(secret.Data["notification-email"])), nil
+}
+
+func updateUserNotificationsEmails(oldAlertmanagerConfig string, newUserNotificationEmails string) string {
+
+	userNotificationregex := regexp.MustCompile(`- name: 'user-notifications'\n  email_configs:\n  - to: '.*com'`)
+	userNotificationMatch := userNotificationregex.FindAllString(oldAlertmanagerConfig, -1)
+
+	emailsRegex := regexp.MustCompile(`'.*com'`)
```
Comment (crobby) [inline pkg/controller/kfdef/kfdef_controller.go:768]: Maybe I'm just lacking some context here...what's up with the ".*com" stuff?

**Comment by anishasthana** [comment]
Lateste live build: quay.io/anishasthana/rhods-operator-live-catalog:1.101.0-1974

File: pkg/controller/kfdef/kfdef_controller.go (hunk 2 above)
Comment (anishasthana) [reply pkg/controller/kfdef/kfdef_controller.go:768]: So the first regex matches the following
```
    - name: 'user-notifications'
      email_configs:
      - to: 'santa@presents.com,elf@presents.com '
```

The second regex matches `santa@presents.com,elf@presents.com` and replaces it with the new emails

**Review by lucferbux — APPROVED** [review]
/lgtm

**Review by VaishnaviHire — COMMENTED** [review]
/lgtm

Worked well when updated the emails through UI

**Comment by jgarciao** [comment]
lgtm too!

**Pull request merged**